segments/helpers.py

```diff
@@ -157,10 +157,10 @@ def execute_raw_user_query(sql):
         result = cursor.fetchall() or []
         total = len(result)
 
-        # Guardrail: Try to ensure results are integers
+        # Guardrail: Filter out None results
         if total > 0 and result[0]:
-            if not type(result[0][0]) == int:
-                raise RuntimeError('Query returned non-integer results')
+            filtered_results = [i for i in result if i[0] is not None]
+            return [filtered_results, total]
```

Comment:
wouldn't `total` need to be the length of the results without the None values?

```python
        if len(result)> 0 and result[0]:
            if not type(result[0][0]) == int:
                raise RuntimeError('Query returned non-integer results')
            filtered_results = [i for i in result if i[0] is not None]
            return [filtered_results, len(filtered_results)]
```

or am I misunderstanding what `total` represents?